tests/drivers/sensor/sbs_gauge/src/test_sbs_gauge.c

```diff
@@ -0,0 +1,314 @@
+/*
+ * Copyright (c) 2021 Leica Geosystems AG
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <ztest.h>
+#include <drivers/sensor.h>
+
+const struct device *get_fuel_gauge_device(void)
+{
+	const struct device *dev = DEVICE_DT_GET_ANY(sbs-gauge);
+
+	zassert_true(device_is_ready(dev), "Fuel Gauge not found");
+
+	return dev;
+}
+
+void test_sbs_gauge_getSensorValue(int16_t channel)
+{
+	struct sensor_value value;
+	const struct device *dev = get_fuel_gauge_device();
+
+	zassert_true(sensor_sample_fetch_chan(dev, channel) < 0, "Sample fetch failed");
+	zassert_true(sensor_channel_get(dev, channel, &value) < 0, "Get sensor value failed");
+}
+
+void test_sbs_gauge_getSensorValue_NotSupported(int16_t channel)
```

Comment:
I do not understand the sense of all the *NotSupported functions. Also, if for some reason it is necessary, you can loop over all the channels and skip the right ones with continue.

Reply:
The idea was to test also the "not implemented" sensor channels. I changed the code and a loop is used for this.